README.md

```diff
@@ -1 +1,107 @@
-# Winter Framework
+# Winter - An IoC Container for Java
+
+Winter is an Inversion of Control (IoC) framework for Java, inspired by Spring. It provides annotation-based dependency injection, bean lifecycle management, and a modular architecture for building scalable applications.
+
+## Features
+
+### 1. Dependency Injection (DI)
+- **@Autowired**: Automatically injects dependencies into beans.
+  - **Autowiring Modes:**
+      - **Single constructor injection** (default if only one constructor exists)
+      - **@Autowired constructor injection** (explicitly marked constructors take precedence)
+      - **@Autowired field injection** (directly injects dependencies into fields)
+      - **@Autowired setter/method injection** (less common but supported)
+- **@Qualifier**: Allows specifying which bean to inject when multiple implementations exist.
+
+  - **Qualifier Modes:**
+      - **@Qualifier("beanName")** (by declared name)
```

Comment:
`@Qualifier("beanName")` - is more a qualifier by bean name. 
Qualifier by declared name is when a field of one bean is named the same as one existing bean and explicit Qualifier annotation is not needed in this case.

Reply:
fixed